### PR TITLE
Fix dealer applications and other misc fixes

### PR DESCRIPTION
**Pull request by kitsuta**
The important fix from here is https://github.com/MidwestFurryFandom/rams/pull/326 -- we were completely ignoring almost all validation for dealer applications.

### CODE REVIEW

File: uber/automated_emails.py [hunk 1]
```diff
@@ -290,14 +290,6 @@ def __init__(self, subject, template, filter, ident, **kwargs):
         needs_approval=True,
         ident='dealer_reg_payment_reminder_last_chance')
 
-    MarketplaceEmailFixture(
-        '{EVENT_NAME} Dealer waitlist has been exhausted',
-        'dealers/waitlist_closing.txt',
-        lambda g: g.status == c.WAITLISTED,
-        # query=Group.status == c.WAITLISTED,
-        when=after(c.DEALER_WAITLIST_CLOSED),
-        ident='uber_marketplace_waitlist_exhausted')
```
Comment (EliAndrewC) [inline uber/automated_emails.py]: Is this something we want to delete?

Comment (kitsuta) [reply uber/automated_emails.py]: Yea, this email hasn't been sent in at least two years. It trips the "email ready to send" reminder forever, which was annoying to the people getting those emails.

**Comment by kitsuta** [comment]
This now also fixes another fairly serious bug introduced with promo-code groups where Stripe transactions did not record attendees' names in the description field.

File: uber/automated_emails.py (hunk 1 above)
Comment (EliAndrewC) [reply uber/automated_emails.py]: Good to know!

**Comment by kitsuta** [comment]
Merging this now, as it fixes a DB migration issue that's preventing us from deploying the new servers.

**Pull request merged**